ts/features/bonus/bpd/components/bpdCardComponent/BpdCardComponent.tsx

```diff
@@ -0,0 +1,266 @@
+    fontSize: 24,
+    lineHeight: 35,
+    marginBottom: -8
+  },
+  amountTextUpperFull: { fontSize: 32 },
+  amountTextBasePreview: {
+    fontSize: 16,
+    lineHeight: 32
+  },
+  amountTextUpperPreview: { fontSize: 24 },
+  alignItemsCenter: {
+    alignItems: "center"
+  },
+  justifyContentCenter: {
+    justifyContent: "center"
+  }
+});
+
+type BadgeDefinition = {
+  style: any;
+  label: string;
+};
+
+export const BpdCardComponent: React.FunctionComponent<Props> = (
+  props: Props
+) => {
+  const formatStatusBadge = (): BadgeDefinition => {
+    const { period } = props;
+
+    switch (period.status) {
+      case "Active":
+        return {
+          style: {
+            backgroundColor: IOColors.white
+          },
+          label: I18n.t("bonus.bpd.detail.status.active")
+        };
+      case "Closed":
+        return {
+          style: {
+            backgroundColor: IOColors.black
+          },
+          label: I18n.t("bonus.bpd.detail.status.closed")
+        };
+      case "Inactive":
+        return {
+          style: {
+            backgroundColor: IOColors.aqua
+          },
+          label: I18n.t("bonus.bpd.detail.status.inactive")
+        };
+      default:
+        return {
+          style: {
+            backgroundColor: IOColors.white
+          },
+          label: "-"
+        };
+    }
+  };
+
+  const amount = formatNumberCentsToAmount(
+    props.totalAmount.totalCashback
+  ).split(",");
+
+  const renderFullCard = () => (
+    <View style={[styles.row, styles.spaced]}>
+      <View style={[styles.column, styles.flex2, styles.spaced]}>
+        <View>
+          <H2 weight={"Bold"} color={"white"}>
+            {I18n.t("bonus.bpd.title")}
+          </H2>
+          <H4 color={"white"} weight={"Regular"}>
+            {format(props.period.startDate, "DD MMM YYYY")} -{" "}
+            {format(props.period.endDate, "DD MMM YYYY")}
+          </H4>
+        </View>
+        <View spacer={true} large />
+        <View style={{ height: 32 }}>
+          <View style={[styles.row, { alignItems: "center" }]}>
+            <Text bold={true} white={true} style={styles.amountTextBaseFull}>
+              €
+              <Text white={true} style={styles.amountTextUpperFull}>
+                {amount[0]},
+              </Text>
+              {amount[1]}
+            </Text>
+            <View hspacer={true} small={true} />
+            <IconFont name="io-lucchetto" size={22} color={IOColors.white} />
+          </View>
+          <H5 color={"white"} weight={"Regular"}>
+            {I18n.t("bonus.bpd.detail.card.earned")}
+          </H5>
+        </View>
+      </View>
+      <View style={[styles.column, styles.flex1, styles.spaced]}>
+        <Badge style={[formatStatusBadge().style, styles.badgeBase]}>
+          <Text
+            semibold={true}
+            style={styles.badgeTextBase}
+            dark={props.period.status !== "Closed"}
+          >
+            {formatStatusBadge().label}
```

Comment:
In order to avoid multiple execution of the function `formatStatusBadge` maybe we can use `const badgeFormat = formatStatusBadge()` in `BpdCardComponent` and use `badgeFormat.style` and `badgeFormat.label`.
What you think?

Reply:
absolutely! sounds good